c7n/resources/apigw.py

```diff
@@ -284,6 +290,28 @@ def augment(self, resources):
             results.append(r)
         return results
 
+    def get_resources(self, ids, cache=True):
+        deployment_ids = []
+        client = utils.local_session(
+            self.manager.session_factory).client('apigateway')
+        for id in ids:
+            # if we get stage arn, we pick rest_api_id
+            if 'arn:aws:apigateway' in id:
+                _, ident = id.rsplit(':', 1)
+                parts = ident.split('/', 4)
+                # if we get stage name in arn, use stage_name to get stage information
```

Comment:
You're handling stage names, deployment IDs and ARNs... is that just to be flexible or because there isn't a common ID available across the events you're trying to cover?

Reply:
rest-stage works on deploymentId. 
For below cloudtrail event, we get ARN with arn:partition:apigateway:region::/apis/api-id/stages/stage-name

to get deploymentId, I have to call get-stage and use deploymentId
```      
events:
  - source: apigateway.amazonaws.com
    ids: requestParameters.resourceArn
    event: TagResource
```

Reply:
@ajkerrigan Can you please review the changes?